commit 14b131197d6e39a01160a5168810083ef2035fbb
Author: Kalesh Singh <kaleshsingh@google.com>
Date:   Tue Jan 21 13:31:02 2025 -0800

    Revert "ltp: syscalls: mincore: Iterate vector in kernel page size granule"
    
    This reverts commit 85eab5a3f56c26fa89bde69c50ce55a08b345b9c.
    
    Reason: 16KB x86_64 now emulates the mincore() syscall, so revert the
    tests to their original versions.
    
    Bug: 390505059
    Test: atest vts_ltp_test_x86_64:syscalls.mincore03_64bit#syscalls.mincore03_64bit
    Test: atest vts_ltp_test_x86_64:syscalls.mincore04_64bit#syscalls.mincore04_64bit
    Signed-off-by: Kalesh Singh <kaleshsingh@google.com>
    Change-Id: I2912d065507d456e2349dd8ac897a8b29cb59877

diff --git a/testcases/kernel/syscalls/mincore/mincore03.c b/testcases/kernel/syscalls/mincore/mincore03.c
index c3ef2d203..774fce98b 100644
--- a/testcases/kernel/syscalls/mincore/mincore03.c
+++ b/testcases/kernel/syscalls/mincore/mincore03.c
@@ -16,7 +16,6 @@
 #include <unistd.h>
 #include <sys/mman.h>
 #include "tst_test.h"
-#include "pgsize_helpers.h"
 
 #define NUM_PAGES 3
 
@@ -47,7 +46,7 @@ static void setup(void)
 static void test_mincore(unsigned int test_nr)
 {
 	const struct tcase *tc = &tcases[test_nr];
-	DECLARE_MINCORE_VECTOR(vec, NUM_PAGES);
+	unsigned char vec[NUM_PAGES];
 	int locked_pages;
 	int count, mincore_ret;
 
@@ -59,10 +58,9 @@ static void test_mincore(unsigned int test_nr)
 	if (mincore_ret == -1)
 		tst_brk(TBROK | TERRNO, "mincore failed");
 	locked_pages = 0;
-	for (count = 0; count < nr_pgs_to_nr_kernel_pgs(NUM_PAGES); count++)
+	for (count = 0; count < NUM_PAGES; count++)
 		if (vec[count] & 1)
 			locked_pages++;
-	locked_pages = nr_kernel_pgs_to_nr_pgs(locked_pages);
 
 	if (locked_pages == tc->expected_pages)
 		tst_res(TPASS, "mincore() reports %s", tc->desc);
diff --git a/testcases/kernel/syscalls/mincore/mincore04.c b/testcases/kernel/syscalls/mincore/mincore04.c
index f718301fb..ed0ab7dfa 100644
--- a/testcases/kernel/syscalls/mincore/mincore04.c
+++ b/testcases/kernel/syscalls/mincore/mincore04.c
@@ -20,7 +20,6 @@
 #include <fcntl.h>
 #include <stdlib.h>
 #include "tst_test.h"
-#include "pgsize_helpers.h"
 
 #define NUM_PAGES 3
 
@@ -65,18 +64,17 @@ static int count_pages_in_cache(void)
 {
 	int locked_pages = 0;
 	int count, ret;
-	DECLARE_MINCORE_VECTOR(vec, NUM_PAGES);
+	unsigned char vec[NUM_PAGES];
 
 	TST_CHECKPOINT_WAIT(0);
 
 	ret = mincore(ptr, size, vec);
 	if (ret == -1)
 		tst_brk(TBROK | TERRNO, "mincore failed");
-	for (count = 0; count < nr_pgs_to_nr_kernel_pgs(NUM_PAGES); count++) {
+	for (count = 0; count < NUM_PAGES; count++) {
 		if (vec[count] & 1)
 			locked_pages++;
 	}
-	locked_pages = nr_kernel_pgs_to_nr_pgs(locked_pages);
 
 	TST_CHECKPOINT_WAKE(1);
 	return locked_pages;
